pkg/plugin/processor/builtin/setField.go

```diff
@@ -0,0 +1,105 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builtin
+
+import (
+	"context"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+)
+
+type setField struct {
+	referenceResolver sdk.ReferenceResolver
+	value             string
+
+	sdk.UnimplementedProcessor
+}
+
+func (p *setField) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:        "field.set",
+		Summary:     "set the value of a certain field",
+		Description: "set the value of a certain field, processor only runs on structured data.",
+		Version:     "v1.0",
+		Author:      "Meroxa, Inc.",
+		Parameters: map[string]sdk.Parameter{
+			"field": {
+				Default:     "",
+				Type:        sdk.ParameterTypeString,
+				Description: "the field to be set, use Go template format.",
+				Validations: []sdk.Validation{
+					{
+						Type: sdk.ValidationTypeRequired,
+					},
+				},
+			},
+			"value": {
+				Default:     "",
+				Type:        sdk.ParameterTypeString,
+				Description: "new value to set the field to.",
+				Validations: []sdk.Validation{
+					{
+						Type: sdk.ValidationTypeRequired,
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func (p *setField) Configure(_ context.Context, cfg map[string]string) error {
+	field, ok := cfg["field"]
```

Comment:
Nitpick: it should not be possible to change the value of the position. Conduit itself will complain, but it's also good to check it here.

Reply:
This could be alternatively done by creating a dummy record, resolving the field and trying to set the field to a test string. Then the fact that the position can't be set would only be coded in one place (the reference).

Reply:
sorry I don't think I understand, wdym by "value of the position"?

Reply:
or I can also just add an exclusion validation for `.Position` using paramgen!

Reply:
that's a good idea!